StreamChatClient_v3/Database/DTOs/ChannelDTO.swift

```diff
@@ -95,12 +95,49 @@ extension NSManagedObjectContext {
 extension ChannelDTO {
     static func channelListFetchRequest(query: ChannelListQuery) -> NSFetchRequest<ChannelDTO> {
         let request = NSFetchRequest<ChannelDTO>(entityName: "ChannelDTO")
-        request.sortDescriptors = [.init(key: "cid", ascending: true)] // TODO: sorting from query
+        request.sortDescriptors = query.sort.sortDescriptors
         request.predicate = NSPredicate(format: "ANY queries.filterHash == %@", query.filter.filterHash)
         return request
     }
 }
 
+public extension Array where Element == Sorting {
+    var sortDescriptors: [NSSortDescriptor] {
+        if isEmpty || (count == 1 && first?.field == "updated_at") {
```

Comment:
> (count == 1 && first?.field == "updated_at")

This looks strange. What's the reason behind this?

Reply:
This is default sorting on backend side. So, if you don't sent sorting value it will be `updated_at = last_message_at OR created_at` and the same if you send `updated_at`.

Reply:
I think it should be added to `test_channelPayload_isStoredAndLoadedFromDB` where we test serializing the payload to DTO.

Reply:
I don't get it. What do you want to add to the serialising test?

Reply:
Sorry, wrong place...